src/node_file.cc

```diff
@@ -127,6 +192,37 @@ static inline bool IsInt64(double x) {
   return x == static_cast<double>(static_cast<int64_t>(x));
 }
 
+static Local<Value> EncodeFilename(Environment* env,
+                                  const char* filename,
+                                  enum encoding encoding) {
```

Comment:
Also, is there any way the `Path` constructor and this can be combined into one?